content/docs/guides/neon-twin-full-pg-dump-restore.md

```diff
@@ -0,0 +1,166 @@
+---
+title: Full Twin
+subtitle: Create a full Twin of your production database
+enableTableOfContents: true
+isDraft: false
+updatedOn: '2025-02-06T19:39:12.345Z'
+---
+
+This workflow will create a full Neon Twin using `pg_dump` and `pg_restore`.
+
+<Admonition type="note">
+To use this workflow, you'll need the PostgreSQL connection string for your Neon database. Follow our [Getting Started Guide](/docs/get-started-with-neon/signing-up#sign-up) to learn how.
+</Admonition>
+
+## Create the workflow
+
+To create the Twin workflow in any GitHub-hosted repository:
+
+1. Create a new directory named `.github` at the root of your project.
+2. Inside this directory, create another directory named `workflows`.
+3. Within the `workflows` directory, create a new file named `create-neon-twin.yml`.
+
+```
+.github
+  |-- workflows
+      |-- create-neon-twin.yml
+```
+
+Add the following code to `create-neon-twin.yml`.
+
+```yml
+name: Create Neon Twin
+
+on:
+  schedule:
+    - cron: '0 0 * * *' # Runs at midnight ET (us-east-1)
+  workflow_dispatch:
+
+env:
+  PROD_DATABASE_URL: ${{ secrets.PROD_DATABASE_URL }} # Production or primary database
+  DEV_DATABASE_URL: ${{ secrets.DEV_DATABASE_URL }} # Development database
+  PG_VERSION: '17'
+
+jobs:
+  dump-and-restore:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Install PostgreSQL
+        run: |
+          sudo apt update
+          yes '' | sudo /usr/share/postgresql-common/pgdg/apt.postgresql.org.sh
+          sudo apt install -y postgresql-${{ env.PG_VERSION }}
+
+      - name: Set PostgreSQL binary path
+        run: echo "POSTGRES=/usr/lib/postgresql/${{ env.PG_VERSION }}/bin" >> $GITHUB_ENV
+
+      - name: Dump and restore data
+        run: |
+          $POSTGRES/pg_dump -Fc -f "${{ github.workspace }}/dump-file.bak" "${{ env.PROD_DATABASE_URL }}"
+          $POSTGRES/pg_restore --clean --no-owner --no-acl --if-exists -d "${{ env.DEV_DATABASE_URL }}" "${{ github.workspace }}/dump-file.bak"
+```
+
+## GitHub Action explained
+
+Below is an explanation of each part of the GitHub Action.
+
+### on
+
+- `name`: The name of the Action as it appears in the GitHub UI.
+- `cron`: The [POSIX cron syntax](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/crontab.html#tag_20_25_07) that defines when the Action will run.
+- `workflow_dispatch`: Enables manual triggering through the GitHub UI.
+
+### env
+
+- `PROD_DATABASE_URL`: The PostgreSQL connection string for your production database.
+- `DEV_DATABASE_URL`: The PostgreSQL connection string for your Neon database.
+- `PG_VERSION`: The version of PostgreSQL to install in the Action environment.
```

Comment:
Do we install the Postgres instance in the Action environment to be able to run pg_dump and pg_restore commands? And I was wondering where the dump file is created. Is it created in the Action environment, and does it continued to live there? Are there any database size / dump file size limitations?

Reply:
That's correct. Postgres is installed in the "VM" a GitHub Actions Ubuntu environment. The dump file is temporarily written to disk in the same environment. When the job is complete, the environment is trashed. 

Regarding disk size, to the best of my knowledge, no. There is no mention of disk size limitations in the docs: https://docs.github.com/en/actions/administering-github-actions/usage-limits-billing-and-administration#usage-limits.